src/server/services/wfs/qgswfsdescribefeaturetype.cpp

```diff
@@ -200,46 +200,37 @@ namespace QgsWfs
     {
       QDomElement geomElem = doc.createElement( QStringLiteral( "element" )/*xsd:element*/ );
       geomElem.setAttribute( QStringLiteral( "name" ), QStringLiteral( "geometry" ) );
-      if ( provider->name() == QLatin1String( "ogr" ) )
-      {
-        // because some ogr drivers (e.g. ESRI ShapeFile, GML)
-        // are not able to determine the geometry type of a layer.
-        // we set to GeometryType
-        geomElem.setAttribute( QStringLiteral( "type" ), QStringLiteral( "gml:GeometryPropertyType" ) );
-      }
-      else
+
+      QgsWkbTypes::Type wkbType = layer->wkbType();
+      switch ( wkbType )
       {
-        QgsWkbTypes::Type wkbType = layer->wkbType();
-        switch ( wkbType )
-        {
-          case QgsWkbTypes::Point25D:
-          case QgsWkbTypes::Point:
-            geomElem.setAttribute( QStringLiteral( "type" ), QStringLiteral( "gml:PointPropertyType" ) );
-            break;
-          case QgsWkbTypes::LineString25D:
-          case QgsWkbTypes::LineString:
-            geomElem.setAttribute( QStringLiteral( "type" ), QStringLiteral( "gml:LineStringPropertyType" ) );
-            break;
-          case QgsWkbTypes::Polygon25D:
-          case QgsWkbTypes::Polygon:
-            geomElem.setAttribute( QStringLiteral( "type" ), QStringLiteral( "gml:PolygonPropertyType" ) );
-            break;
-          case QgsWkbTypes::MultiPoint25D:
-          case QgsWkbTypes::MultiPoint:
-            geomElem.setAttribute( QStringLiteral( "type" ), QStringLiteral( "gml:MultiPointPropertyType" ) );
-            break;
-          case QgsWkbTypes::MultiLineString25D:
-          case QgsWkbTypes::MultiLineString:
-            geomElem.setAttribute( QStringLiteral( "type" ), QStringLiteral( "gml:MultiLineStringPropertyType" ) );
-            break;
-          case QgsWkbTypes::MultiPolygon25D:
-          case QgsWkbTypes::MultiPolygon:
-            geomElem.setAttribute( QStringLiteral( "type" ), QStringLiteral( "gml:MultiPolygonPropertyType" ) );
-            break;
-          default:
-            geomElem.setAttribute( QStringLiteral( "type" ), QStringLiteral( "gml:GeometryPropertyType" ) );
-            break;
-        }
+        case QgsWkbTypes::Point25D:
```

Comment:
I'm not familiar with the wfs spec, but should we also be including PointZ, PointM, PointZM here? (And similar for the other types)

Reply:
Added a feature request: https://issues.qgis.org/issues/17095